### ISSUE TITLE
Haxe 3.3.0 + Flash: Illegal write to read-only property

### ISSUE THREAD

**Issue by Gama11** [openfl/openfl#1025]
Compiling any OpenFL project with Haxe 3.3.0 leads to this runtime error at startup on Flash:

```
ReferenceError: Error #1074: Illegal write to read-only property LEFT on class flash.ui.KeyLocation.
	at boot_7e63/init()[C:\HaxeToolkit\haxe\lib\openfl\git\extern\openfl\ui\KeyLocation.hx:14]
	at flash::Boot/start()[C:\HaxeToolkit\haxe\std\flash\Boot.hx:69]
	at boot_7e63()[C:\HaxeToolkit\haxe\std\flash\Boot.hx:38]
```

I don't think this is related to the fixes in #1022, `KeyLocation` wasn't touched there.

**Comment by Simn** [comment]
That looks like it's trying to do a static init on that field. The question is why it would do that.

**Comment by Gama11** [comment]
Because [the `KeyLocation` extern](https://github.com/openfl/openfl/blob/develop/extern/openfl/ui/KeyLocation.hx#L14) uses an abstract enum.

**Comment by Simn** [comment]
Ah yes, combining that with `@:native` is not a good idea.

**Comment by Gama11** [comment]
Looks like @jgranick fixed this in 56dd9f27497b12b96b81874aa3bac657317978e6.
